In [1]:
import pandas as pd
#import warnings
#warnings.filterwarnings("ignore")

In [2]:
file_to_load = "vine_table.csv"

In [4]:
df = pd.read_csv(file_to_load)
df

review_id  star_rating  helpful_votes  total_votes vine  \
0         RTIS3L2M1F5SM            5              0            0    N   
1         R1ZV7R40OLHKD            5              0            0    N   
2        R3BH071QLH8QMC            1              0            1    N   
3        R127K9NTSXA2YH            3              0            0    N   
4        R32ZWUXDJPW27Q            4              0            0    N   
...                 ...          ...            ...          ...  ...   
1785992   RPC430LWZJ60T            5              1            1    N   
1785993  R347MZT5FH6HRJ            5              2            2    N   
1785994   RJ5BETZP0VIUS            1              0            2    N   
1785995   R85QTDO2KZMGO            1              1            3    N   
1785996  R2W36RZZZNTYWF            5              1            1    N   

        verified_purchase  
0                       Y  
1                       Y  
2                       Y  
3                       Y  
4                       Y  
...                   ...  
1785992                 N  
1785993                 N  
1785994                 N  
1785995                 N  
1785996                 N  

[1785997 rows x 6 columns]

In [5]:
df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1785997 entries, 0 to 1785996
Data columns (total 6 columns):
 #   Column             Dtype 
---  ------             ----- 
 0   review_id          object
 1   star_rating        int64 
 2   helpful_votes      int64 
 3   total_votes        int64 
 4   vine               object
 5   verified_purchase  object
dtypes: int64(3), object(3)
memory usage: 81.8+ MB


### Filter by total votes

In [6]:
filtered_vine_df = df[df['total_votes'] >= 20]
filtered_vine_df.head()

review_id  star_rating  helpful_votes  total_votes vine  \
55    R4PKAZRQJJX14            1             21           34    N   
74   R2CI0Y288CC7E2            1             21           35    N   
209  R127WEQY2FM1T3            1            147          175    N   
289  R3EZ0EPYLDA34S            1             14           31    N   
483  R2FJ94555FZH32            2             55           60    N   

    verified_purchase  
55                  N  
74                  Y  
209                 Y  
289                 Y  
483                 N

In [9]:
new_vine_df = filtered_vine_df.loc[filtered_vine_df.helpful_votes / filtered_vine_df.total_votes >= 0.5]
new_vine_df.head()

review_id  star_rating  helpful_votes  total_votes vine  \
55    R4PKAZRQJJX14            1             21           34    N   
74   R2CI0Y288CC7E2            1             21           35    N   
209  R127WEQY2FM1T3            1            147          175    N   
483  R2FJ94555FZH32            2             55           60    N   
537  R1U3AR67RE273L            1             51           65    N   

    verified_purchase  
55                  N  
74                  Y  
209                 Y  
483                 N  
537                 Y

## Reviews written as part of the vine program

In [11]:
vine_reviews = new_vine_df[new_vine_df['vine'] == 'Y']
vine_reviews.head()

review_id  star_rating  helpful_votes  total_votes vine  \
32611   R3KKUSGFZWSUIY            5             56           63    Y   
33112   R10FO5UKKVZBK2            3             23           23    Y   
69680    RM4KSGEOR7MU1            5             19           24    Y   
155361   RG7VRMYLEXD23            4             22           26    Y   
239327  R11O4YSCPSNL6L            3             20           26    Y   

       verified_purchase  
32611                  N  
33112                  N  
69680                  N  
155361                 N  
239327                 N

## Reviews not written as part of the vine program

In [13]:
not_vine_reviews = new_vine_df[new_vine_df['vine'] == 'N']
not_vine_reviews.head()

review_id  star_rating  helpful_votes  total_votes vine  \
55    R4PKAZRQJJX14            1             21           34    N   
74   R2CI0Y288CC7E2            1             21           35    N   
209  R127WEQY2FM1T3            1            147          175    N   
483  R2FJ94555FZH32            2             55           60    N   
537  R1U3AR67RE273L            1             51           65    N   

    verified_purchase  
55                  N  
74                  Y  
209                 Y  
483                 N  
537                 Y

## 5-Star Review Analysis

In [15]:
count_vine = len(vine_reviews)
count_vine

94

In [16]:
count_nonvine = len(not_vine_reviews)
count_nonvine

40471

In [17]:
max_rating_vine = len(vine_reviews[vine_reviews['star_rating']==5])
max_rating_vine

48

In [18]:
max_rating_nonvine = len(not_vine_reviews[not_vine_reviews['star_rating']==5])
max_rating_nonvine

15663

In [19]:
max_rating_vine/count_vine

0.5106382978723404

In [20]:
max_rating_nonvine/count_nonvine

0.38701786464381904

## Results:
* Vine (paid) Reviews
1. 94 total reviews
2. 48 were 5 star reviews
3. **51%** of vine (paid) reviews were 5 star
* Unpaid Reviews
1. 40,471 total reviews
2. 15663 5 star reviews
3. 38.7% of unpaid reviews were 5 star

## Further Analysis
Average star rating of reviews per program (paid and unpaid
* Paid Vine Program

In [21]:
round(vine_reviews['star_rating'].mean(),2)

4.2

* Unpaid Program


In [22]:
round(not_vine_reviews['star_rating'].mean(),2)


3.35